rusk/src/lib/prover.rs

```diff
@@ -53,11 +50,11 @@ pub fn verify_proof(tx: &Transaction) -> Result<bool> {
             .map(|_| CircuitOutput::ZERO_COMMITMENT.into()),
     );
 
-    let keys = match inputs.len() {
-        1 => keys_for(ExecuteCircuitOneTwo::circuit_id())?,
-        2 => keys_for(ExecuteCircuitTwoTwo::circuit_id())?,
-        3 => keys_for(ExecuteCircuitThreeTwo::circuit_id())?,
-        4 => keys_for(ExecuteCircuitFourTwo::circuit_id())?,
+    let circuit_profile = match inputs.len() {
+        1 => CircuitProfile::from_name("ExecuteCircuitOneTwo")?,
+        2 => CircuitProfile::from_name("ExecuteCircuitTwoTwo")?,
+        3 => CircuitProfile::from_name("ExecuteCircuitThreeTwo")?,
+        4 => CircuitProfile::from_name("ExecuteCircuitFourTwo")?,
```

Comment:
Can this be improved caching the derived verifier?
So to not execute `from_name` every time a transaction is gonna be accepted into the mempool